### PR TITLE
DatePicker blog post and docs updates

### PR DESCRIPTION
**Pull request by devongovett**
For release.

### CODE REVIEW

File: packages/dev/docs/pages/blog/datepicker.mdx [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+```tsx snippet
+<Flex direction="column" alignItems="center">
+  <DateField label="Birth date" />
+</Flex>
+```
+
+This approach has benefits for internationalization and accessibility, as well as usability on mobile. For internationalization, individual segments avoid the problem of parsing dates in various formats entirely. The date format is automatically determined based on the locale, and the user only needs to fill in the values and not worry about messing up the separators or getting the order wrong. Each segment is also individually labeled for accessibility, so users always know which field they are on (e.g. “year”, “month”, “day”, etc.). This is much easier to use for screen reader users than a plain text field where the expected format is unknown. Finally, on mobile, we can take advantage of the numeric software keyboard, which is nicer to use than a full QUERTY keyboard.
```
Comment (snowystinger) [inline packages/dev/docs/pages/blog/datepicker.mdx:61]: I'm going to assume you meant QWERTY ;P

File: packages/dev/docs/pages/blog/datepicker.mdx [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+date.set({ month: 11 }, 'later');
+// => 2020-11-01T01:00:00-08:00[America/Los_Angeles]
+```
+
+See [the docs](../../internationalized/date/ZonedDateTime.html#setting-fields) for more details on how this works.
+
+## Conclusion
+
+Correctly manipulating dates and times in *really hard*. Making assumptions about calendar systems, time zones, locales, date and time arithmetic, etc. is a recipe for bugs when users around the world interact with your app. [@internationalized/date](../../internationalized/date/index.html) provides a library of objects and functions that help handle these differences and allow you to manipulate dates from all users consistently. It is a completely independent library, so even if you aren’t using React Aria, React Spectrum, or even React, you can still take advantage of it for all your date and time manipulation needs!
```
Comment (majornista) [inline packages/dev/docs/pages/blog/datepicker.mdx:241]: Correctly manipulating dates and times is *really hard*.

Comment (majornista) [reply packages/dev/docs/pages/blog/datepicker.mdx:241]: "is" not "in"

**Review by majornista — CHANGES_REQUESTED** [review]

File: packages/dev/docs/pages/blog/datepicker.mdx [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+```tsx
+import {toCalendar, HebrewCalendar, GregorianCalendar} from '@internationalized/date';
+
+let hebrewDate = new CalendarDate(new HebrewCalendar(), 5781, 1, 1);
+let gregorianDate = toCalendar(hebrewDate, new GregorianCalendar());
+gregorianDate.toString();
+// => '2020-09-19'
+```
+
+Most developers are not aware of all of the minutiae of the different calendar systems. As a result, it’s easy to make assumptions in applications based on your own knowledge, which may lead to bugs for users in other parts of the world. To make this easier, the APIs for all of our date and time picker components handle all of the details and calendar conversions for you. If you provide a date in the Gregorian calendar system as a value to a component, that’s what you’ll get back, no matter which calendar system the user is interacting with. This allows applications to deal with dates from all users consistently, even if users enter dates in a different calendar system than they are stored in.
```
Comment (snowystinger) [inline packages/dev/docs/pages/blog/datepicker.mdx:192]: condescending tone, maybe consider changing to
```suggestion
It’s easy to make wrong assumptions in applications due to the complexity and subtle differences in calendar systems, which may lead to bugs for users in other parts of the world. To make this easier, the APIs for all of our date and time picker components handle all of the details and calendar conversions for you. If you provide a date in the Gregorian calendar system as a value to a component, that’s what you’ll get back, no matter which calendar system the user is interacting with. This allows applications to deal with dates from all users consistently, even if users enter dates in a different calendar system than they are stored in.
```

File: packages/dev/docs/pages/blog/datepicker.mdx [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+Another important area we considered with our calendar components was mobile. With touch screen readers, users access each control on screen using swipe gestures to move a virtual cursor forward and backward. Because of this, calendars can be quite tedious to navigate because they contain so many elements, especially when multiple months are displayed at once. We made sure to provide context when a user enters a calendar of the whole range of visible dates, and included an extra visually hidden “next” button at the end of the dates so a user doesn’t need to swipe all the way back to the start to navigate to the next month. The column headers are also skipped to improve ease of navigation, since the day names are already included in the label of each cell.
+
+<Video src={calendarMobileScreenReader + '#t=0.1'} preload="metadata" controls style={{maxHeight: '600px', maxWidth: 'min(100%, 1920px)', display: 'block', margin: '40px auto'}} />
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/dev/docs/pages/blog/datepicker.mdx:106]: I did have the VO captions turned on, they appear at the bottom.

[the remark this answers is not in the document]
Comment (majornista) [reply packages/dev/docs/pages/blog/datepicker.mdx:106]: Perhaps hard to see under video controls?

[the remark this answers is not in the document]
Comment (devongovett) [reply packages/dev/docs/pages/blog/datepicker.mdx]: Updated to use the video you sent me with the captions. I tried to figure out how to use WebVTT captions but couldn't get it to work.

File: packages/dev/docs/pages/blog/datepicker.mdx [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+- **Internationalization** – We have extensive support for internationalization, including 13 different calendar systems such as Gregorian, Buddhist, Islamic, Persian, and more. Locale-specific formatting, number systems, 12 and 24 hour time, and right-to-left support are available as well.
+- **Accessibility** – All of our date and time picker components have been tested across desktop and mobile devices, and with many different input methods including mouse, touch, and keyboard. We have worked hard to ensure screen reader announcements are clear and consistent.
+- **Customizability** – As with all of React Aria, our hooks give you full control over the rendering and styling of your components, while letting us handle the internationalization and accessibility challenges for you. We have examples using many different styling libraries, such as Tailwind CSS, Styled Components, CSS modules, and Chakra UI.
+
+<Video src={datepickerVideo} muted loop autoPlay style={{maxWidth: 'min(100%, 1920px)', display: 'block', margin: '40px auto'}} />
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/dev/docs/pages/blog/datepicker.mdx]: Good idea! Made it not autoplay if that media query is true. Also made it possible to pause the autoplaying video even when the media query is not true.

[the remark this answers is not in the document]
Comment (majornista) [reply packages/dev/docs/pages/blog/datepicker.mdx]: Are there other autoPlay videos in blog posts that we could treat the same?

File: packages/dev/docs/pages/blog/datepicker.mdx [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+
+  heroVideo.onclick = toggle;
+  heroVideo.onkeydown = e => {
+    if (e.key === ' ' || e.key === 'Enter') {
+      e.preventDefault();
+      toggle();
+    }
+  }
+}
+```
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/dev/docs/pages/blog/datepicker.mdx]: added to client.js so it applies to all pages automatically.

File: packages/dev/docs/pages/blog/datepicker.mdx [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+```tsx
+import {toCalendar, HebrewCalendar, GregorianCalendar} from '@internationalized/date';
+
+let hebrewDate = new CalendarDate(new HebrewCalendar(), 5781, 1, 1);
+let gregorianDate = toCalendar(hebrewDate, new GregorianCalendar());
+gregorianDate.toString();
+// => '2020-09-19'
+```
+
+Most developers are not aware of all of the minutiae of the different calendar systems. As a result, it’s easy to make assumptions in applications based on your own knowledge, which may lead to bugs for users in other parts of the world. To make this easier, the APIs for all of our date and time picker components handle all of the details and calendar conversions for you. If you provide a date in the Gregorian calendar system as a value to a component, that’s what you’ll get back, no matter which calendar system the user is interacting with. This allows applications to deal with dates from all users consistently, even if users enter dates in a different calendar system than they are stored in.
```
Comment (majornista) [inline packages/dev/docs/pages/blog/datepicker.mdx]: Nitpick. The last sentence ends with an unnecessary dangling preposition.
```suggestion
Most developers are not aware of all of the minutiae of the different calendar systems. As a result, it’s easy to make assumptions in applications based on your own knowledge, which may lead to bugs for users in other parts of the world. To make this easier, the APIs for all of our date and time picker components handle all of the details and calendar conversions for you. If you provide a date in the Gregorian calendar system as a value to a component, that’s what you’ll get back, no matter which calendar system the user is interacting with. This allows applications to deal with dates from all users consistently, even if users enter dates in a different calendar system than that in which they are stored.
```

File: packages/dev/docs/pages/blog/date-and-time-pickers-for-all.mdx [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+```tsx
+import {toCalendar, HebrewCalendar, GregorianCalendar} from '@internationalized/date';
+
+let hebrewDate = new CalendarDate(new HebrewCalendar(), 5781, 1, 1);
+let gregorianDate = toCalendar(hebrewDate, new GregorianCalendar());
+gregorianDate.toString();
+// => '2020-09-19'
+```
+
+It’s easy to make assumptions based on the calendar system you use every day and test with during development, which may lead to bugs when users in other parts of the world use your app. To make this easier, the APIs for all of our date and time picker components handle all of the details and calendar conversions for you. If you provide a date in the Gregorian calendar system as a value to a component, that’s what you’ll get back, no matter which calendar system the user is interacting with. This allows applications to deal with dates from all users consistently, even if users enter dates in a different calendar system than they are stored in.
```
Comment (majornista) [inline packages/dev/docs/pages/blog/date-and-time-pickers-for-all.mdx:215]: Nitpick. Last sentence in this paragraph ends with a dangling preposition.
```suggestion
It’s easy to make assumptions based on the calendar system you use every day and test with during development, which may lead to bugs when users in other parts of the world use your app. To make this easier, the APIs for all of our date and time picker components handle all of the details and calendar conversions for you. If you provide a date in the Gregorian calendar system as a value to a component, that’s what you’ll get back, no matter which calendar system the user is interacting with. This allows applications to deal with dates from all users consistently, even if users enter dates in a different calendar system than that in which they are stored.
```

**Review by majornista — DISMISSED** [review]

**Review by majornista — DISMISSED** [review]

**Review by LFDanLu — APPROVED** [review]
Small comments for things that I noticed when reading the post, but minor enough that I'm happy approving this. Super well polished, excited for this to go out!

File: packages/dev/docs/pages/blog/date-and-time-pickers-for-all.mdx [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+- **Internationalization** – We have extensive support for internationalization, including 13 different calendar systems such as Gregorian, Buddhist, Islamic, Persian, and more. Locale-specific formatting, number systems, 12 and 24 hour time, and right-to-left support are available as well.
+- **Accessibility** – All of our date and time picker components have been tested across desktop and mobile devices, and with many different input methods including mouse, touch, and keyboard. We have worked hard to ensure screen reader announcements are clear and consistent.
+- **Customizability** – As with all of React Aria, our hooks give you full control over the rendering and styling of your components, while letting us handle the internationalization and accessibility challenges for you. We have examples using many different styling libraries, such as Tailwind CSS, Styled Components, CSS modules, and Chakra UI.
+
+<Video
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/dev/docs/pages/blog/date-and-time-pickers-for-all.mdx:42]: Yeah this one doesn't have `controls`. I felt it was less important than the others, as this is really just showing a looping demo reel where the others are demonstrating something specific.

**Pull request merged**